### PR TITLE
[nuklear] Upgrade to latest nuklear (12/05/2022)

### PR DESCRIPTION
**Pull request by SamuelMarks**
**Describe the pull request**
Upgrade port

- #### What does your PR fix?
Upgrades the port

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
All

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/nuklear/vcpkg.json [hunk 1]
```diff
@@ -1,9 +1,10 @@
 {
   "name": "nuklear",
-  "version-date": "2021-03-18",
+  "version-date": "2022-05-12",
   "port-version": 1,
```
Comment (Thomas1664) [inline ports/nuklear/vcpkg.json:4]:
```suggestion
  "version-date": "2022-05-12",
```

**Review by Thomas1664 — CHANGES_REQUESTED** [review]

File: ports/nuklear/portfile.cmake [hunk 2]
```diff
@@ -1,12 +1,13 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
-    REPO Immediate-Mode-UI/Nuklear
-    REF 6e80e2a646f35be4afc157a932f2936392ec8f74 # accessed on 2021-04-04
-    SHA512 ce064dff721111749d4056717879f42d3e24bb94655dd2b04c137eb7391d2c90d0b1b95155912c100b537f74fd150aedc48e0ac85eb72963c66e35ac81048323
-    HEAD_REF master
+    REPO            Immediate-Mode-UI/Nuklear
+    REF             0a77211844458673446aa99b8a0ae1090a04e413 # 2022-05-12
+    SHA512          57ad5578c534d19ce6dd5e9b8c7b725da6e7f379a5289daed6b7128a1671dc5c368175d1e780a7fcb3405fad989e3bc50a8a921d62301e64e58014e351845cae
+    HEAD_REF        master
```
Comment (Adela0814) [inline ports/nuklear/portfile.cmake:6]: Please remove the extra space

File: ports/nuklear/portfile.cmake [hunk 3]
```diff
@@ -15,8 +16,8 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
 )
 
 vcpkg_cmake_configure(
-    SOURCE_PATH ${SOURCE_PATH}
-    OPTIONS ${FEATURE_OPTIONS}
+    SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS     "${FEATURE_OPTIONS}"
```
Comment (Adela0814) [inline ports/nuklear/portfile.cmake:20]: Please remove the extra space.

**Review by Adela0814 — CHANGES_REQUESTED** [review]

File: ports/nuklear/portfile.cmake (hunk 2 above)
Comment (SamuelMarks) [reply ports/nuklear/portfile.cmake:6]: A few other ports also have the extra space to align all the text, makes it more readable [IMHO]; is this a blocker?

File: ports/nuklear/portfile.cmake (hunk 3 above)
Comment (SamuelMarks) [reply ports/nuklear/portfile.cmake:20]: A few other ports also have the extra space to align all the text, makes it more readable [IMHO]; is this a blocker?

Comment (Adela0814) [reply ports/nuklear/portfile.cmake:20]: This is not a blocker, just a suggestion. For other ports with extra spaces, we will delete them when updating the port. This is to unify with our other port formats.

Comment (dg0yt) [reply ports/nuklear/portfile.cmake:20]: There should be no double quotes around `${FEATURE_OPTIONS}`. It is to be expanded as a list of values, not to be used as a single string value.

**Review by Adela0814 — DISMISSED** [review]

**Comment by ras0219-msft** [comment]
Thanks for this PR!

However, we prefer to not take large nonessential formatting changes like this -- imagine if another user had a differently-configured autoformatter and every update the file was thrashing back and forth between different formats!

Please apply the following patch, then this PR LGTM:

```diff
diff --git a/ports/nuklear/CMakeLists.txt b/ports/nuklear/CMakeLists.txt
index 89e344165..eba91e1fd 100644
--- a/ports/nuklear/CMakeLists.txt
+++ b/ports/nuklear/CMakeLists.txt
@@ -6,33 +6,29 @@ option(INSTALL_DEMO "Install the demo code" OFF)
 
 add_library(nuklear INTERFACE)
 
-include(GNUInstallDirs)
-target_include_directories(nuklear INTERFACE "$<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}/nuklear>")
+target_include_directories(nuklear INTERFACE $<INSTALL_INTERFACE:include/nuklear>)
 target_compile_definitions(nuklear INTERFACE NK_IMPLEMENTATION)
 
 # Installation
 install(
     TARGETS nuklear
     EXPORT unofficial-nuklear
-    ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-    LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}"
-    RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
+    RUNTIME DESTINATION bin
+    LIBRARY DESTINATION lib
+    ARCHIVE DESTINATION lib
 )
 
-install(FILES "${CMAKE_CURRENT_LIST_DIR}/nuklear.h"
-        DESTINATION "${CMAKE_INSTALL_INCLUDEDIR}/nuklear")
+install(FILES "${CMAKE_CURRENT_LIST_DIR}/nuklear.h" DESTINATION "include/nuklear")
 
 
-install(EXPORT unofficial-nuklear
-        FILE "unofficial-nuklear-config.cmake"
-        DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/unofficial-nuklear")
+install(EXPORT unofficial-nuklear FILE unofficial-nuklear-config.cmake DESTINATION share/unofficial-nuklear)
 
 if (INSTALL_EXAMPLE)
-    install(DIRECTORY "${CMAKE_CURRENT_LIST_DIR}/example"
-            DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/nuklear")
+    install(DIRECTORY "${CMAKE_CURRENT_LIST_DIR}/example" DESTINATION share/nuklear)
+
 endif()
 
 if (INSTALL_DEMO)
-    install(DIRECTORY "${CMAKE_CURRENT_LIST_DIR}/demo"
-            DESTINATION "${CMAKE_INSTALL_DATAROOTDIR}/nuklear")
+    install(DIRECTORY "${CMAKE_CURRENT_LIST_DIR}/demo" DESTINATION share/nuklear)
+
 endif()
diff --git a/ports/nuklear/portfile.cmake b/ports/nuklear/portfile.cmake
index 4ca2095c1..f30f54d0f 100644
--- a/ports/nuklear/portfile.cmake
+++ b/ports/nuklear/portfile.cmake
@@ -1,13 +1,12 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
-    REPO            Immediate-Mode-UI/Nuklear
-    REF             0a77211844458673446aa99b8a0ae1090a04e413 # 2022-05-12
-    SHA512          57ad5578c534d19ce6dd5e9b8c7b725da6e7f379a5289daed6b7128a1671dc5c368175d1e780a7fcb3405fad989e3bc50a8a921d62301e64e58014e351845cae
-    HEAD_REF        master
+    REPO Immediate-Mode-UI/Nuklear
+    REF 0a77211844458673446aa99b8a0ae1090a04e413 # 2022-05-12
+    SHA512 57ad5578c534d19ce6dd5e9b8c7b725da6e7f379a5289daed6b7128a1671dc5c368175d1e780a7fcb3405fad989e3bc50a8a921d62301e64e58014e351845cae
+    HEAD_REF master
 )
 
-file(COPY "${CURRENT_PORT_DIR}/CMakeLists.txt"
-     DESTINATION "${SOURCE_PATH}")
+file(COPY "${CURRENT_PORT_DIR}/CMakeLists.txt" DESTINATION "${SOURCE_PATH}")
 
 vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
     FEATURES
@@ -16,8 +15,8 @@ vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
 )
 
 vcpkg_cmake_configure(
-    SOURCE_PATH "${SOURCE_PATH}"
-    OPTIONS     ${FEATURE_OPTIONS}
+    SOURCE_PATH ${SOURCE_PATH}
+    OPTIONS ${FEATURE_OPTIONS}
 )
 
 vcpkg_cmake_install()
@@ -27,6 +26,4 @@ vcpkg_cmake_config_fixup(PACKAGE_NAME unofficial-nuklear)
 
 file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug")
 
-file(INSTALL "${SOURCE_PATH}/src/LICENSE"
-     DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}"
-     RENAME copyright)
+file(INSTALL "${SOURCE_PATH}/src/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
diff --git a/ports/nuklear/vcpkg.json b/ports/nuklear/vcpkg.json
index 25261916e..1e48043bf 100644
--- a/ports/nuklear/vcpkg.json
+++ b/ports/nuklear/vcpkg.json
@@ -1,9 +1,9 @@
 {
   "name": "nuklear",
-  "version-date": "2022-05-12",
+  "version-date": "2021-03-18",
+  "port-version": 1,
   "description": "This is a minimal state immediate mode graphical user interface toolkit written in ANSI C and licensed under public domain",
   "homepage": "https://github.com/Immediate-Mode-UI/Nuklear",
-  "license": "Unlicense OR MIT",
   "dependencies": [
     {
       "name": "vcpkg-cmake",
diff --git a/versions/baseline.json b/versions/baseline.json
index 3c94378e7..87bf58967 100644
--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -4921,8 +4921,8 @@
       "port-version": 0
     },
     "nuklear": {
-      "baseline": "2022-05-12",
-      "port-version": 0
+      "baseline": "2021-03-18",
+      "port-version": 1
     },
     "numactl": {
       "baseline": "2.0.14",
diff --git a/versions/n-/nuklear.json b/versions/n-/nuklear.json
index 9286743e6..b0784005d 100644
--- a/versions/n-/nuklear.json
+++ b/versions/n-/nuklear.json
@@ -6,7 +6,7 @@
       "port-version": 0
     },
     {
-      "git-tree": "9c243e51fbb0d9fa4562a3e4a8eba9a7fb2e0cdd",
+      "git-tree": "d0f86ceb1fe400a4e2cdd6ff76d82dfe539a0203",
       "version-date": "2021-03-18",
       "port-version": 1
     },
```

**Comment by SamuelMarks** [comment]
Wait what are all those non-whitespace changes?

**Comment by ras0219-msft** [comment]
Sorry, I had some errors in that patch as well as perhaps it didn't fully post correctly. I've pushed a branch instead:
https://github.com/ras0219-msft/vcpkg/commit/5858af159a63fb9f88d68182221e67e3ceef4d4e.

In addition to reverting the whitespace changes, I've also reverted many of the changes to the `CMakeLists.txt`. While your changes would be following best practices for an external project CMake buildsystem, we intend replacement buildsystems inside vcpkg to be as direct and concrete as possible -- that means there's no need for `GNUInstallDirs` etc because vcpkg has a fixed install format.

Hopefully your upstream PR (https://github.com/Immediate-Mode-UI/Nuklear/pull/462) gets merged, we can delete this entire file, and never look back :)

```sh
git fetch https://github.com/ras0219-msft/vcpkg dev/roschuma/pr24829
git merge FETCH_HEAD
```

**Comment by PhoebeHui** [comment]
@SamuelMarks, could you please apply the patch?

**Comment by Adela0814** [comment]
Pinging @SamuelMarks for response.

**Comment by SamuelMarks** [comment]
Hmm, I thought I applied the patch, just double checked:
```
/vcpkg (nuklear-12_05_2022-upgrade) $ git fetch https://github.com/ras0219-msft/vcpkg dev/roschuma/pr24829
From https://github.com/ras0219-msft/vcpkg
 * branch                dev/roschuma/pr24829 -> FETCH_HEAD
/vcpkg (nuklear-12_05_2022-upgrade) $ git pull
Already up to date.
/vcpkg (nuklear-12_05_2022-upgrade) $ git merge FETCH_HEAD
Already up to date.
/vcpkg (nuklear-12_05_2022-upgrade) $ git push
Everything up-to-date
```

Even triple-checked:
```
/vcpkg (nuklear-12_05_2022-upgrade) $ git merge 5858af159a63fb9f88d68182221e67e3ceef4d4e
Already up to date.
```

**Comment by SamuelMarks** [comment]
Hmm, I thought I applied the patch, just double checked:
```sh
/vcpkg (nuklear-12_05_2022-upgrade) $ git fetch https://github.com/ras0219-msft/vcpkg dev/roschuma/pr24829
From https://github.com/ras0219-msft/vcpkg
 * branch                dev/roschuma/pr24829 -> FETCH_HEAD
/vcpkg (nuklear-12_05_2022-upgrade) $ git pull
Already up to date.
/vcpkg (nuklear-12_05_2022-upgrade) $ git merge FETCH_HEAD
Already up to date.
/vcpkg (nuklear-12_05_2022-upgrade) $ git push
Everything up-to-date
```

Even triple-checked:
```sh
/vcpkg (nuklear-12_05_2022-upgrade) $ git merge 5858af159a63fb9f88d68182221e67e3ceef4d4e
Already up to date.
```

**Review by Adela0814 — APPROVED** [review]

**Pull request merged**